client/implementation/src/main/java/io/smallrye/graphql/client/impl/GraphQLClientsConfiguration.java

```diff
@@ -55,40 +60,32 @@ public GraphQLClientsConfiguration() {
         }
         for (String clientName : detectedClientNames) {
             GraphQLClientConfiguration configuration = new GraphQLClientConfiguration();
-            configuration.setUrl(mpConfig.getValue(clientName + "/mp-graphql/url", String.class));
-            configuration.setHeaders(getConfigurationValueMap(clientName, "header", mpConfig));
+            mpConfig.getOptionalValue(clientName + "/mp-graphql/url", String.class)
+                    .ifPresent(configuration::setUrl);
+            configuration.setHeaders(getConfiguredHeaders(clientName, mpConfig));
             clients.put(clientName, configuration);
         }
     }
 
     /**
-     * Scan the passed Java interfaces for `@GraphQLClientApi` annotations and create and register
-     * client configuration objects for them. This needs to be called by the runtime some time during initialization,
+     * Scan the passed Java interface for `@GraphQLClientApi` annotations and create and register
+     * client configuration objects for it. This needs to be called by the runtime some time during initialization,
      * before clients are actually created.
      */
-    public void addTypesafeClientApis(List<Class<?>> apis) {
-        if (apis == null) {
-            SmallRyeGraphQLClientLogging.log.apisNotSet();
-            apis = Collections.emptyList();
-        }
```

Comment:
This could never have happened

Reply:
Not with pure SmallRye code, but this method is called by Quarkus so I think this should stay guarded

Reply:
done